packages/utils/src/ratelimit.ts

```diff
@@ -85,7 +85,16 @@ export function updateRateLimits(
         updatedRateLimits.all = now + delay;
       } else {
         for (const category of categories.split(';')) {
-          updatedRateLimits[category] = now + delay;
+          if (category === 'metric_bucket') {
+            const namespaces = limit.split(':', 5)[4] ? limit.split(':', 5)[4].split(';') : null;
+
+            if (!namespaces || namespaces.includes('custom')) {
+              // back off transmitting metrics from the SDK
+              updatedRateLimits[category] = now + delay;
```

Comment:
this is the same as line 96, can we merge the logic somehow?

Reply:
I added all in one `if` statement. But as it is not so clear anymore that `'metric_bucket'` actually is the category when `namespaces` is defined, I added a comment that explains it :)